File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.js
/**
 * @param {number[]} plantTime
 * @param {number[]} growTime
 * @return {number}
 */
const earliestFullBloom = function(plantTime, growTime) {
    const n = growTime.length;
    const days = growTime.map((g, i) => [plantTime[i], g]).sort((a, b) => b[1] - a[1]);
    
    let max = 0, sum = 0;
    
    days.forEach(([plant, grow]) => {
        sum += plant;
        max = Math.max(max, sum + grow);
    })
    
    return max;
};

File: hard/2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.cs
public class Solution {
    public int EarliestFullBloom(int[] plantTime, int[] growTime) {
        int n = growTime.Length;
        int[][] days = new int[n][];
        
        for(int i = 0; i < n; i++)
        {
            days[i] = new int[2] { plantTime[i], growTime[i] };
        }
        
        days = days.OrderByDescending(x => x[1]).ToArray();
        
        int max = 0, plantSum = 0;
        
        for(int i = 0; i < n; i++)
        {
            int plant = days[i][0], grow = days[i][1];
            plantSum += plant;
            max = Math.Max(max, plantSum + grow);
        }
        
        return max;
    }
}